pkg/agent/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,439 @@
+	c.supportBundleCollectionsMutex.Lock()
+	defer c.supportBundleCollectionsMutex.Unlock()
+
+	c.supportBundleCollections[supportbundle.Name] = supportbundle
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) deleteSupportBundleCollection(supportbundle *cpv1b2.SupportBundleCollection) {
+	c.supportBundleCollectionsMutex.Lock()
+	defer c.supportBundleCollectionsMutex.Unlock()
+
+	delete(c.supportBundleCollections, supportbundle.Name)
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting", "controllerName", controllerName)
+	defer klog.InfoS("Shutting down", "controllerName", controllerName)
+
+	go wait.NonSlidingUntil(c.watchSupportBundleCollections, 5*time.Second, stopCh)
+
+	go wait.Until(c.worker, time.Second, stopCh)
+	<-stopCh
+}
+
+func (c *SupportBundleController) GetSupportBundleCollectionCondition() v1alpha1.SupportBundleCollectionConditionType {
+	c.conditionLock.RLock()
+	defer c.conditionLock.RUnlock()
+	return c.collectionCondition
+}
+
+func (c *SupportBundleController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *SupportBundleController) processNextWorkItem() bool {
+	obj, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	if key, ok := obj.(string); !ok {
+		klog.Errorf("Expected string in work queue but got %#v", obj)
+		return true
+	} else if err := c.syncSupportBundleCollection(key); err == nil {
+		klog.InfoS("Successfully synced support bundle", "name", key)
+	} else {
+		// Skip retrying as the time may not meet the requirements for SupportBundle.
+		klog.ErrorS(err, "Error syncing SupportBundleCollection", "name", key)
+	}
+	return true
+}
+
+func (c *SupportBundleController) syncSupportBundleCollection(key string) error {
+	klog.InfoS("Sync support bundle collection", "name", key)
+	c.supportBundleCollectionsMutex.RLock()
+	defer c.supportBundleCollectionsMutex.RUnlock()
+	supportBundle, exists := c.supportBundleCollections[key]
+	if !exists {
+		return nil
+	}
+	err := c.generateSupportBundle(supportBundle)
+	if err != nil {
+		c.updateSupportBundleCollectionCondition(v1alpha1.CollectionFailure)
+		if updateErr := c.updateSupportBundleCollectionStatus(key, false, err); updateErr != nil {
+			return fmt.Errorf("failed to update failed collection status: %w", updateErr)
+		}
+		return fmt.Errorf("failed to generate support bundle: %w", err)
+	}
+	c.updateSupportBundleCollectionCondition(v1alpha1.CollectionCompleted)
+	if updateErr := c.updateSupportBundleCollectionStatus(key, true, err); updateErr != nil {
+		return fmt.Errorf("failed to update complete collection status: %w", updateErr)
+	}
+
+	return nil
+}
+
+func (c *SupportBundleController) generateSupportBundle(supportBundle *cpv1b2.SupportBundleCollection) error {
+	klog.V(2).InfoS("Generate support bundle collection", "name", supportBundle.Name)
```

Comment:
```suggestion
	klog.V(2).InfoS("Generating support bundle collection", "name", supportBundle.Name)
```
https://github.com/tnqn/code-review-comments#follow-logging-convention, same suggestion applies to several logs.